### PR TITLE
Ext function form

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add function form
- Add `columns` property to `FormGroup`
- Add `language` property to `CodeEditor`
- Handle `min` and `max` props in `NumberInput`
- Handle `enum` translations
- Add quick fix for default values for variables, to be done properly in #1835 
- Fix RuleInput bug ("Add all" would insert `core-api-group` instead of `''`).

### CODE REVIEW

**Review by Lyczeq — CHANGES_REQUESTED** [review]

**Comment by Wawrzyn321** [comment]
I'll also add Env variables to the form.

File: core-ui/src/components/Extensibility/components-form/NumberRenderer.js [hunk 1]
```diff
@@ -18,10 +18,14 @@ export function NumberRenderer({
   placeholder,
   ...props
 }) {
-  const { tFromStoreKeys, t: tExt } = useGetTranslation();
+  const { tFromStoreKeys, tryTranslate } = useGetTranslation();
 
   const schemaPlaceholder = schema.get('placeholder');
 
+  const numberProps = Object.fromEntries(
+    ['min', 'max'].map(prop => [prop, schema.get(prop)]),
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core-ui/src/components/Extensibility/components-form/NumberRenderer.js:26]: [`633e07a` (#1842)](https://github.com/kyma-project/busola/pull/1842/commits/633e07ac0a6540ce292ee0ea3e92e415cec19f17)

File: core-ui/src/components/Extensibility/helpers/index.js [hunk 2]
```diff
@@ -69,6 +69,7 @@ export const useGetTranslation = path => {
       // change 'undefined' to undefined etc.
       return unstringify(translation);
     },
+    tryTranslate: path => (exists(path) ? t(path) : ''),
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core-ui/src/components/Extensibility/helpers/index.js]: I rolled back those changes, they aren't needed anymore.

File: docs/extensibility/form-section.md [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 The **form** section contains a list of objects that define which fields are included in the final form. All given fields are placed in the advanced form by default. It's possible to add a field to the simple form by providing the `simple: true` flag. You can also remove it from the advanced form by providing the `advanced: false` flag.
 
-Any parameters that are not handled by the widget are added to the schema directly, so it's possible to add or override existing values. For example, add an **enum** parameter to provide selectable values in a field.
+Any parameters that are not handled by the widget are added to the schema directly, so it's possible to add or override existing values. For example, add an **enum** parameter to provide selectable values in a field or specify additional parameters to improve the schema validation, e.g. `min` and `max` attributes for numeric inputs to enable HTML validation.
```
Comment (grego952) [inline docs/extensibility/form-section.md:22]:
```suggestion
Any parameters that are not handled by the widget are added to the schema directly, so it's possible to add or override existing values. For example, add an **enum** parameter to provide selectable values in a field or specify additional parameters to improve the schema validation, for example, `min` and `max` attributes for numeric inputs to enable HTML validation.
```

File: docs/extensibility/form-section.md [hunk 4]
```diff
@@ -195,22 +195,28 @@ CodeEditor widgets render a versatile code editor that can be used to edit any v
 
 #### Widget-specific parameters
 
+- **language** - a jsonata expression resolving the desired language. It has access to `$root` variable, containing the entire resource.
```
Comment (grego952) [inline docs/extensibility/form-section.md:198]:
```suggestion
- **language** - a JSONata expression resolving the desired language. It has access to the `$root` variable, containing the entire resource.
```

File: docs/extensibility/form-section.md [hunk 5]
```diff
@@ -195,22 +195,28 @@ CodeEditor widgets render a versatile code editor that can be used to edit any v
...
 
 ```json
 {
+  "path": "spec.my-data",
+  "widget": "CodeEditor",
   "path": "spec.data",
   "widget": "CodeEditor",
   "inputInfo": "Data needs to be a valid JSON object.",
-  "description": "Data that will be passed on to the application."
+  "description": "Data that will be passed on to the application.",
```
Comment (grego952) [inline docs/extensibility/form-section.md:211]:
```suggestion
  "description": "Data is passed on to the application.",
```

File: docs/extensibility/form-section.md [hunk 6]
```diff
@@ -195,22 +195,28 @@ CodeEditor widgets render a versatile code editor that can be used to edit any v
...
+  "language": "'JSON'"
 }
 ```
 
 <img src="./assets/form-widgets/CodeEditor.png" alt="Example of a code editor widget" style="border: 1px solid #D2D5D9">
 
+> NOTE: Remember to put both single and double quotes if you want to use plain language (e.g. "'YAML'"). Specyfing just double quotes ("YAML") makes Busola try to access nonexistent `YAML` variable, resulting the language to be `undefined`.
```
Comment (grego952) [inline docs/extensibility/form-section.md:218]:
```suggestion
> NOTE: Remember to put both single and double quotes if you want to use plain language (for example, "'YAML'"). Specifying just double quotes ("YAML") makes Busola try to access a nonexistent `YAML` variable, resulting in the language being `undefined`.
```

File: docs/extensibility/form-section.md [hunk 7]
```diff
@@ -341,6 +347,10 @@ Presentation widgets do not handle data directly and only serve to group content
 
 FormGroup widgets render an `object` as a collapsible section.
 
+#### Widget-specific parameters
+
+- **columns** - number of columns the content should be rendered in. Defaults to 1.
```
Comment (grego952) [inline docs/extensibility/form-section.md:352]:
```suggestion
- **columns** - number of columns the content is rendered in. Defaults to 1.
```

**Review by Lyczeq — DISMISSED** [review]

**Review by grego952 — DISMISSED** [review]

**Review by Lyczeq — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**